flake-module.nix

```diff
@@ -159,9 +202,22 @@ in
                     hlint;
                 };
                 buildTools = lib.attrValues (defaultBuildTools hp // cfg.buildTools hp);
-                package = cfg.modifier (hp.callCabal2nixWithOptions cfg.name cfg.root "" { });
+                package =
+                  if cfg.topLevelPackages.enableMultiPkgs then
+                    hp.${cfg.topLevelPackages.defaultPkg}
+                  else
+                    cfg.modifier (hp.callCabal2nixWithOptions cfg.name cfg.root "" { });
                 devShell = with pkgs.haskell.lib;
-                  (addBuildTools package buildTools).envFunc { withHoogle = true; };
+                  (addBuildTools package buildTools).envFunc (
```

Comment:
Perhaps instead of `envFunc`, `hp.shellFor` would be nicer, because it merges the environments of multiple packages.